src/lib/createStardustComponent.tsx

```diff
@@ -0,0 +1,29 @@
+import createComponentInternal from './createComponent'
+import * as React from 'react'
+import { pick } from 'lodash'
+import { ComponentSlotClasses } from '@stardust-ui/react'
+
+export interface RenderStardustResultConfig {
+  classes: ComponentSlotClasses
+  rtl: boolean
+}
+
+export interface CreateStardustComponentConfig<P> {
+  displayName?: string
+  render: (props: P & { stardust: RenderStardustResultConfig }) => React.ReactNode
+}
+
+const createComponent = <P extends {} = {}, S extends {} = {}>({
+  displayName = 'StardustComponent',
+  render,
+}: CreateStardustComponentConfig<P>): React.SFC<P> => {
+  return createComponentInternal<P, S>({
+    displayName,
+    render: (config, props) => {
+      const filteredConfig = pick(config, ['classes', 'rtl'])
+      return render(Object.assign({ stardust: filteredConfig }, props))
```

Comment:
I had to use here Object.assing as spread is not supported for generics :\ https://github.com/Microsoft/TypeScript/issues/10727

Reply:
Yep, it's a known issue